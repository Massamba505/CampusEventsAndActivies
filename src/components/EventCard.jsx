import truncateString from '../utils/truncate';

const EventCard = ({ event, onGetTickets }) => {
  const { title, description, eventAuthor:organizer, date, location, images } = event;

  return (
    <div className="rounded-lg max-h-max max-w-md border bg-white p-6 shadow-lg transition-transform transform hover:scale-105">
      <div className="flex flex-col h-full items-start space-y-4">
        <img
          src={images[0]?images[0]:" "}
          alt="Event"
          className="mx-auto aspect-video rounded-xl object-cover w-full h-48"
          width="300"
          height="200"
        />
        <div className="flex-1 flex flex-col space-y-2 max-h-64">
          <h3 className="text-xl font-bold text-gray-800">
            {truncateString(title, 80)}
          </h3>
          <p className="text-gray-600">
            {truncateString(description, 120)}
          </p>
          
          <div className="flex mt-3 items-center space-x-2">
            <OrganizerIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{organizer}</small>
          </div>
          <div className="flex mt-2 items-center space-x-2">
            <CalendarDaysIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{date}</small>
          </div>
          <div className="flex mt-2 items-center space-x-2">
            <LocateIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{location}</small>
          </div>
          <div className="mt-3 w-full">
            <button
              onClick={onGetTickets}
              className="self-start inline-flex h-9 items-center justify-center rounded-md bg-indigo-600 px-4 py-2 text-sm font-medium text-white shadow hover:bg-indigo-700 transition-colors"
            >
              Get Tickets
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};




function CalendarDaysIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M8 2v4" />
      <path d="M16 2v4" />
      <rect width="18" height="18" x="3" y="4" rx="2" />
      <path d="M3 10h18" />
      <path d="M8 14h.01" />
      <path d="M12 14h.01" />
      <path d="M16 14h.01" />
      <path d="M8 18h.01" />
      <path d="M12 18h.01" />
      <path d="M16 18h.01" />
    </svg>
  )
}

// function CalendarIcon(props) {
//   return (
//     <svg
//       {...props}
//       xmlns="http://www.w3.org/2000/svg"
//       width="24"
//       height="24"
//       viewBox="0 0 24 24"
//       fill="none"
//       stroke="currentColor"
//       strokeWidth="2"
//       strokeLinecap="round"
//       strokeLinejoin="round"
//     >
//       <path d="M8 2v4" />
//       <path d="M16 2v4" />
//       <rect width="18" height="18" x="3" y="4" rx="2" />
//       <path d="M3 10h18" />
//     </svg>
//   )
// }



function OrganizerIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M12 20v-6M6 20v-6M18 20v-6M2 10h20" />
    </svg>
  );
}

function LocateIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <line x1="2" x2="5" y1="12" y2="12" />
      <line x1="19" x2="22" y1="12" y2="12" />
      <line x1="12" x2="12" y1="2" y2="5" />
      <line x1="12" x2="12" y1="19" y2="22" />
      <circle cx="12" cy="12" r="7" />
    </svg>
  )
}

export default EventCard;
