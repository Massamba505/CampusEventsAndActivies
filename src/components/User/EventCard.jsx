import { useState } from 'react';
import truncateString from '../../utils/truncate';
import EditEventModal from './EditModal';
import DeleteEvent from './DeleteEvent';
import { XIcon } from 'lucide-react';

const EventCard = ({ event, onDeleteEvent, onEditEvent }) => {
  const {
    event_id,
    title,
    eventAuthor,
    date,
    startTime,
    endTime,
    location,
    images = [],
    isPaid,
    ticketPrice,
    maxAttendees,
    currentAttendees,
    food_stalls,
    isCancelled = false,
    discount = null,
    category = [],
  } = event;

  const [modalVisible, setModalVisible] = useState(false); // Control modal visibility
  const [modalVisibleD, setModalVisibleD] = useState(false); // Control modal visibility
  const [eventId, setEventId] = useState(false); // Control modal visibility

  // Function to handle edit button click
  const handleEditClick = (id) => {
    setEventId(id)
    setModalVisible(true); // Open modal
  };

  // Function to handle event update in modal
  const handleUpdate = () => {
    onEditEvent(); // Refresh event after an update
  };

  // Function to handle event deletion
  const handleDelete = () => {
    setModalVisibleD(false);
    onDeleteEvent(event_id); // Trigger event delete
  };

  return (
    <div className='relative text-decoration-none bg-white'>
    <div className="relative rounded-lg w-80 border p-2 transition-transform transform hover:bg-gray-100">
      {/* <XIcon className='absolute z-10 right-1/2 text-red-500 w-full h-full'/> */}
      <div className="flex flex-col h-full items-start space-y-4">
        {/* Image Section */}
        <div className='relative flex justify-center w-full'>
          <div className="relative">
            <img className="mx-auto aspect-video rounded-xl object-cover w-full h-40" src={images[0]} alt={title} />
            <p className={`absolute top-0 ${!isPaid ? "bg-blue-600" : "bg-green-600"} text-white font-semibold py-1 px-3 rounded-br-lg rounded-tl-lg`}>
              {!isPaid ? "FREE" : `$${ticketPrice}`}
            </p>
            {discount > 0 && (
              <p className="absolute top-0 right-0 bg-yellow-300 text-gray-800 font-semibold py-1 px-3 rounded-tr-lg rounded-bl-lg">
                {discount}%
              </p>
            )}
          </div>
        </div>
  
        {/* Info Section */}
        <div className="flex-1 flex flex-col w-full space-y-2">
          {/* Title */}
          <h3 className="text-xl font-bold text-gray-800">
            {truncateString(title, 30)}
          </h3>
  
          {/* Organizer */}
          <div className="flex mt-1 items-center space-x-2">
            <OrganizerIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{eventAuthor}</small>
          </div>
  
          {/* Date and Time */}
          <div className="flex mt-2 items-center space-x-2">
            <CalendarDaysIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{date} {startTime} - {endTime}</small>
          </div>
  
          {/* Location */}
          <div className="flex mt-2 items-center space-x-2">
            <LocateIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{location}</small>
          </div>
  
          {/* Categories */}
          <div className="flex mt-2 items-center space-x-2">
            <CategoryIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">Categories: {category?.map((cat) => cat.name).join(', ') || 'N/A'}</small>
          </div>
  
          {/* Attendees */}
          <div className="flex mt-2 items-center space-x-2">
            <AttendeesIcon className="h-4 w-4 text-gray-500" />
            <small className="text-sm text-gray-500">{currentAttendees}/{maxAttendees || 'Unlimited'} Attendees</small>
          </div>
  
          {/* Food Stalls */}
          {food_stalls && (
            <div className="flex mt-2 items-center space-x-2">
              <FoodIcon className="h-4 w-4 text-gray-500" />
              <small className="text-sm text-gray-500">Food Stalls Available</small>
            </div>
          )}
  
          {/* Ticket Info */}
          <div className="flex mt-2 items-center space-x-2">
            {isCancelled ? (
              <>
                <XIcon className="h-4 w-4 text-red-500" />
                <small className="text-sm text-red-500 font-bold">Event Canceled</small>
              </>
            ) : (
              <>
                <TicketIcon className="h-4 w-4 text-green-600" />
                <small className={`text-sm font-bold text-green-600`}>
                  Ticket Price: ${ticketPrice}
                </small>
              </>
            )}
          </div>
  
          {/* Action Buttons */}
          {
            !isCancelled?(
              <div className="mt-3 w-full flex space-x-2">
                <button onClick={()=>{handleEditClick(event_id)}} className="self-start inline-flex justify-center text-xl w-full bg-yellow-500 hover:bg-yellow-600 text-white py-2 rounded-xl shadow-md transition duration-200">
                  Edit
                </button>
                <button onClick={()=>{setModalVisibleD(true)}} className="self-start inline-flex justify-center text-xl w-full bg-red-500 hover:bg-red-600 text-white py-2 rounded-xl shadow-md transition duration-200">
                  Cancel Event
                </button>
              </div>
            ):(
              <></>
            )
          }
        </div>
      </div>
    </div>
  
    {/* Edit Event Modal */}
    {modalVisibleD && (
      <DeleteEvent
        modalVisible={modalVisibleD}
        setModalVisible={setModalVisibleD}
        onDeleteEvent={handleDelete}
      />
    )}
    {/* Edit Event Modal */}
    {modalVisible && (
      <EditEventModal
        modalVisible={modalVisible}
        setModalVisible={setModalVisible}
        eventId={eventId} // Pass the current event data to the modal
        onUpdate={handleUpdate}
      />
    )}
  </div>
  
  );
};

// Icon Components

function CalendarDaysIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M8 2v4" />
      <path d="M16 2v4" />
      <rect width="18" height="18" x="3" y="4" rx="2" />
      <path d="M3 10h18" />
    </svg>
  );
}

function OrganizerIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M12 20v-6M6 20v-6M18 20v-6M2 10h20" />
    </svg>
  );
}

function LocateIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <line x1="2" x2="5" y1="12" y2="12" />
      <line x1="19" x2="22" y1="12" y2="12" />
      <line x1="12" x2="12" y1="2" y2="5" />
      <line x1="12" x2="12" y1="19" y2="22" />
      <circle cx="12" cy="12" r="7" />
    </svg>
  );
}


function CategoryIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <rect x="2" y="7" width="20" height="14" rx="2" ry="2" />
      <path d="M16 3h4v4M8 3h8v4M2 3h4v4" />
    </svg>
  );
}

function AttendeesIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <circle cx="12" cy="7" r="4" />
      <path d="M5.5 20.5a8.38 8.38 0 0 1 13 0" />
    </svg>
  );
}

function FoodIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M3 12h18M3 6h18M5 6v16M19 6v16" />
      <path d="M3 6h18v4M8 6v12" />
    </svg>
  );
}

function TicketIcon(props) {
  return (
    <svg
      {...props}
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <rect x="3" y="8" width="18" height="8" rx="2" ry="2" />
      <path d="M8 12h2M14 12h2M4 12h.01M20 12h.01" />
    </svg>
  );
}

export default EventCard;
