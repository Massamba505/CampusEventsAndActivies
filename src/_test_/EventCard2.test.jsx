import { render, screen,fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import EventCard from '../components/User/EventCard'; // Adjust the import path as necessary
import '@testing-library/jest-dom';

describe('User EventCard Component', () => {
    const mockEvent = {
      event_id: '1',
      title: 'Sample Event',
      eventAuthor: 'John Doe',
      date: '23/10/2024',
      startTime: '10:00',
      endTime: '14:00',
      location: '123 Main St',
      images: ['https://example.com/image.jpg'],
      isPaid: true,
      ticketPrice: 20,
      maxAttendees: 100,
      currentAttendees: 50,
      food_stalls: true,
      isCancelled: false,
      discount: 10,
      category: [{ name: 'Music' }, { name: 'Art' }],
    };
  
    const onDeleteEventMock = vi.fn();
    const onEditEventMock = vi.fn();
  
    beforeEach(() => {
      onDeleteEventMock.mockClear();
      onEditEventMock.mockClear();
    });
  
    it('renders event details correctly', () => {
      render(<EventCard event={mockEvent} onDeleteEvent={onDeleteEventMock} onEditEvent={onEditEventMock} />);
  
      // Check if the event card is in the document
      const eventCard = screen.getByTestId('event-card');
      expect(eventCard).toBeInTheDocument();
  
      // Check event title
      expect(screen.getByText(mockEvent.title)).toBeInTheDocument();
  
      // Check event author
      expect(screen.getByText(mockEvent.eventAuthor)).toBeInTheDocument();
  
      // Check date and time
      expect(screen.getByText(`${mockEvent.date} ${mockEvent.startTime} - ${mockEvent.endTime}`)).toBeInTheDocument();
  
      // Check location
      expect(screen.getByText(mockEvent.location)).toBeInTheDocument();
  
      // Check ticket price
      expect(screen.getByText(`Ticket Price: $${mockEvent.ticketPrice}`)).toBeInTheDocument();
  
      // Check categories
      expect(screen.getByText(`Categories: ${mockEvent.category.map(cat => cat.name).join(', ')}`)).toBeInTheDocument();
    });
  
    it('triggers delete event when Cancel Event button is clicked', () => {
      render(<EventCard event={mockEvent} onDeleteEvent={onDeleteEventMock} onEditEvent={onEditEventMock} />);
  
      // Click the Cancel Event button
      const cancelButton = screen.getByRole('button', { name: /Cancel Event/i });
      fireEvent.click(cancelButton);
  
      // The deletion modal should open (assuming DeleteEvent handles the modal logic)
      // Simulating modal confirmation (assuming it's handled in the DeleteEvent component)
      const confirmDeleteButton = screen.getByRole('button', { name: /Cancel Event/i }); // Change this to the actual button name in your DeleteEvent component
      fireEvent.click(confirmDeleteButton);
  
      expect(onDeleteEventMock).toHaveBeenCalledWith(mockEvent.event_id);
    });
  
    // it('opens edit modal on Edit button click', () => {
    //   render(<EventCard event={mockEvent} onDeleteEvent={onDeleteEventMock} onEditEvent={onEditEventMock} />);
  
    //   // Click the Edit button
    //   const editButton = screen.getByRole('button', { name: /Edit/i });
    //   fireEvent.click(editButton);
  
    //   // Verify the edit modal opens (check some property in the modal or call to onEditEvent)
    //   expect(onEditEventMock).toHaveBeenCalled();
    // });
  
    it('shows canceled event message if isCancelled is true', () => {
      const canceledEvent = { ...mockEvent, isCancelled: true };
  
      render(<EventCard event={canceledEvent} onDeleteEvent={onDeleteEventMock} onEditEvent={onEditEventMock} />);
  
      // Check if the canceled message is displayed
      expect(screen.getByText(/event canceled/i)).toBeInTheDocument();
    });
  
  });
