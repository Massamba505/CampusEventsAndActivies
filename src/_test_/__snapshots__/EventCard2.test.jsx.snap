// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EventCard Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative text-decoration-none bg-white"
  >
    <div
      class="relative rounded-lg sm:w-80 border p-2 transition-transform transform hover:bg-gray-100"
    >
      <div
        class="flex flex-col h-full items-start space-y-4"
      >
        <div
          class="relative flex justify-center w-full"
        >
          <div
            class="relative"
          >
            <img
              class="mx-auto aspect-video rounded-xl object-cover w-full h-40"
            />
            <p
              class="absolute top-0 bg-blue-600 text-white font-semibold py-1 px-3 rounded-br-lg rounded-tl-lg"
            >
              FREE
            </p>
          </div>
        </div>
        <div
          class="flex-1 flex flex-col w-full space-y-2"
        >
          <h3
            class="text-xl font-bold text-gray-800"
          />
          <div
            class="flex mt-1 items-center space-x-2"
          >
            <svg
              class="lucide lucide-user h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            />
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              2024-10-23  - 
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="2"
                x2="5"
                y1="12"
                y2="12"
              />
              <line
                x1="19"
                x2="22"
                y1="12"
                y2="12"
              />
              <line
                x1="12"
                x2="12"
                y1="2"
                y2="5"
              />
              <line
                x1="12"
                x2="12"
                y1="19"
                y2="22"
              />
              <circle
                cx="12"
                cy="12"
                r="7"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            />
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="20"
                x="2"
                y="7"
              />
              <path
                d="M16 3h4v4M8 3h8v4M2 3h4v4"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              Categories: N/A
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="7"
                r="4"
              />
              <path
                d="M5.5 20.5a8.38 8.38 0 0 1 13 0"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              /Unlimited Attendees
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-green-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="8"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="8"
              />
              <path
                d="M8 12h2M14 12h2M4 12h.01M20 12h.01"
              />
            </svg>
            <small
              class="text-sm font-bold text-green-600"
            >
              Ticket Price: $
            </small>
          </div>
          <div
            class="mt-3 w-full flex space-x-2"
          >
            <button
              class="self-start inline-flex justify-center text-base sm:text-xl w-full bg-yellow-500 hover:bg-yellow-600 text-white py-2 rounded-xl shadow-md transition duration-200"
            >
              Edit
            </button>
            <button
              class="self-start inline-flex justify-center sm:text-xl w-full bg-red-500 hover:bg-red-600 text-white py-2 rounded-xl shadow-md transition duration-200"
            >
              Cancel Event
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`User EventCard Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative text-decoration-none bg-white"
  >
    <div
      class="relative rounded-lg sm:w-80 border p-2 transition-transform transform hover:bg-gray-100"
    >
      <div
        class="flex flex-col h-full items-start space-y-4"
      >
        <div
          class="relative flex justify-center w-full"
        >
          <div
            class="relative"
          >
            <img
              alt="Sample Event"
              class="mx-auto aspect-video rounded-xl object-cover w-full h-40"
              src="https://example.com/image.jpg"
            />
            <p
              class="absolute top-0 bg-green-600 text-white font-semibold py-1 px-3 rounded-br-lg rounded-tl-lg"
            >
              $20
            </p>
            <p
              class="absolute top-0 right-0 bg-yellow-300 text-gray-800 font-semibold py-1 px-3 rounded-tr-lg rounded-bl-lg"
            >
              10%
            </p>
          </div>
        </div>
        <div
          class="flex-1 flex flex-col w-full space-y-2"
        >
          <h3
            class="text-xl font-bold text-gray-800"
          >
            Sample Event
          </h3>
          <div
            class="flex mt-1 items-center space-x-2"
          >
            <svg
              class="lucide lucide-user h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              John Doe
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              23/10/2024 10:00 - 14:00
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="2"
                x2="5"
                y1="12"
                y2="12"
              />
              <line
                x1="19"
                x2="22"
                y1="12"
                y2="12"
              />
              <line
                x1="12"
                x2="12"
                y1="2"
                y2="5"
              />
              <line
                x1="12"
                x2="12"
                y1="19"
                y2="22"
              />
              <circle
                cx="12"
                cy="12"
                r="7"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              123 Main St
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="20"
                x="2"
                y="7"
              />
              <path
                d="M16 3h4v4M8 3h8v4M2 3h4v4"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              Categories: Music, Art
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="7"
                r="4"
              />
              <path
                d="M5.5 20.5a8.38 8.38 0 0 1 13 0"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              50/100 Attendees
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-gray-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 12h18M3 6h18M5 6v16M19 6v16"
              />
              <path
                d="M3 6h18v4M8 6v12"
              />
            </svg>
            <small
              class="text-sm text-gray-500"
            >
              Food Stalls Available
            </small>
          </div>
          <div
            class="flex mt-2 items-center space-x-2"
          >
            <svg
              class="h-4 w-4 text-green-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="8"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="8"
              />
              <path
                d="M8 12h2M14 12h2M4 12h.01M20 12h.01"
              />
            </svg>
            <small
              class="text-sm font-bold text-green-600"
            >
              Ticket Price: $20
            </small>
          </div>
          <div
            class="mt-3 w-full flex space-x-2"
          >
            <button
              class="self-start inline-flex justify-center text-base sm:text-xl w-full bg-yellow-500 hover:bg-yellow-600 text-white py-2 rounded-xl shadow-md transition duration-200"
            >
              Edit
            </button>
            <button
              class="self-start inline-flex justify-center sm:text-xl w-full bg-red-500 hover:bg-red-600 text-white py-2 rounded-xl shadow-md transition duration-200"
            >
              Cancel Event
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
